### PR TITLE
fix(submission): add special parsing for array columns in tsv

### PR DESCRIPTION
**Pull request by kulgan**

### CODE REVIEW

File: sheepdog/utils/transforms/__init__.py [hunk 1]
```diff
@@ -240,7 +239,7 @@ def convert_type(to_cls, key, value, enforce_int_file_size=False):
             elif value_type == list:
                 if isinstance(value, str):
                     # FIXME: Maybe we should have a different separator
```
Comment (kulgan) [inline sheepdog/utils/transforms/__init__.py:241]: :eyes:

File: sheepdog/utils/transforms/__init__.py [hunk 2]
```diff
@@ -290,3 +289,51 @@ def set_reader(self, doc):
         doc = '\n'.join(strip(doc).splitlines())
         f = StringIO(doc)
         self.reader = csv.DictReader(f, delimiter=',')
+
+
+def parse_tsv_array(raw):
```
Comment (philtom-ctds) [inline sheepdog/utils/transforms/__init__.py:294]: We should use the `csv` module here.  By doing that, we get consistent, well-defined behavior. In addition, it gives us all the existing csv behavior for free.  For example, if we ever add quotes to an array element, we can escape it with two quotes, which is common behavior that the `csv` module supports.

It'd be something like:
```
import csv
import io

if not raw:
    return []

reader = csv.reader(io.StringIO(raw))
return next(reader)
```

File: sheepdog/utils/transforms/__init__.py (hunk 1 above)
Comment (philtom-ctds) [reply sheepdog/utils/transforms/__init__.py:241]: We can remove this comment.

File: sheepdog/utils/transforms/__init__.py (hunk 2 above)
Comment (kulgan) [reply sheepdog/utils/transforms/__init__.py:294]: I tried this for a bit, but found some weird behavior before switching to this implementation. Quoted strings needs to escaped in the tsv file using extra quotes, the first tsv reader unescape them so the next DictReader (the one you suggested) will see them as normal string and produce weird result.

Comment (anmaxvl) [reply sheepdog/utils/transforms/__init__.py:294]: so if I get it right: `"foo","bar","baz"` becomes `foo","bar","baz` or something?

**Pull request merged**